File: f2py_src/interp/interp.py
from f_interp import interp
import numpy


interp.speed_test()






"""
def f(x):
  return 0.03 * x**4 - 1.63 * x**3 + 1.22 * x**2 - 0.0001 * x + 3.7
  
def g(x):
  return 2.3 - 1.2 * x + 0.2 * x**2
  
  
x = numpy.linspace(0,10, 101)
y = f(x[:])


print(x)
print(y)

xi = 2.773
yi = interp.interpolate(xi, x, y)

print(xi, yi, f(xi))



x = numpy.linspace(0,10, 11)
y = f(x[:])

print(x)
print(y)

xi = 0
yi = interp.trap(xi, x, y)
print(xi, yi)

xi = 5
yi = interp.trap(xi, x, y)
print(xi, yi)

xi = 10
yi = interp.trap(xi, x, y)
print(xi, yi)

xi = 5.5
yi = interp.trap(xi, x, y)
print(xi, yi)


x = numpy.linspace(0,10,11)
y = f(x[:])
fill = interp.fill(x, y, 21,4)


print(x)
print(y)
print(fill)
"""

File: f2py_src/spline/src/interp/interp.f90
!#############################################################
!#  HOW TO USE:
!#
!#
!# interp.interp4(x, x_points, y_points, y)               # Estimates f(x) at x    4 data points
!# interp.interp4dydx(x, x_points, y_points, y)           # Estimates f'(x) at x   4 data points
!# interp.interp4dydx(x, x_points, y_points, y)           # Estimates f^n(x) at x  4 data points
!#
!# interp.interpn(x, x_points, y_points, y)               # Estimates f(x) at x    n data points
!# interp.interp4dydx(x, x_points, y_points, y)           # Estimates f'(x) at x   n data points
!# interp.interp4dydx(x, x_points, y_points, y)           # Estimates f^p(x) at x  n data points
!#
!# interp.interpolate(xi, x, y, n_interp_in, yi)          # Searches large dataset, takes smaller subset and interpolates
!# interp.trap(xi, x, y, yi)                              # Very simple, searches dataset, line between 2 cloest points 
!#                                                        # used e.g. cross section data, not very well behaved data
!#
!# interp.fill(x, y, arr_l, arr_w, n_interp_in, out_arr)  # Takes data x, y and changes size in new array
!#                                                        # columns of new array x, f(x), f'(x), f''(x)...and so on
!#
!# fill_zoor(x, y, arr_l, arr_w,                          # Same as fill, but sets min/max for x
!#           x_min_range, x_max_range,                    # fills with zero out side of this range
!#           n_interp_in, out_arr)
!#                                                        
!#
!#############################################################

MODULE interp

USE kinds

IMPLICIT NONE

INCLUDE "interp.globals.f90"
INCLUDE "interp.interfaces.f90"

CONTAINS

INCLUDE "interp.interp4.f90"
INCLUDE "interp.interp4dydx.f90"
INCLUDE "interp.interp4dydxn.f90"

INCLUDE "interp.interpn.f90"
INCLUDE "interp.interpndydx.f90"
INCLUDE "interp.interpndydxn.f90"


INCLUDE "interp.interpolate.f90"
INCLUDE "interp.trap.f90"

INCLUDE "interp.fill.f90"
INCLUDE "interp.fill_zoor.f90"

END MODULE interp
